xray/segment.go

```diff
@@ -136,14 +136,17 @@ func BeginSegmentWithSampling(ctx context.Context, name string, r *http.Request,
 		seg.Dummy = true
 	}
 
-	// create a new context to close it on segment close;
-	// this makes sure the closing of the segment won't affect the client's code, that uses the returned context
-	ctx1, cancelCtx := context.WithCancel(ctx)
-	seg.cancelCtx = cancelCtx
-	go func() {
-		<-ctx1.Done()
-		seg.handleContextDone()
-	}()
+	// don't allocate resources for dummy segments
+	if !seg.Dummy {
+		// create a new context to close it on segment close;
+		// this makes sure the closing of the segment won't affect the client's code, that uses the returned context
```

Comment:
I think we can improve on the original wording, Does this makes sense to you? Let me know!

```suggestion
	// Dummy segments don't get sent and don't need a goroutine to cancel them.
	if !seg.Dummy {
		// Create a new context for to cancel segment.
		// Start new goroutine to listen for segment close/cancel events.
		// Cancel simply calls `segment.Close()`.
		//
		// This way, even if the client consumes the `ctx.Done()` event, the
		// X-Ray SDK has a way of canceling (closing) the segment in the
		// `segment.Close()` method using this new cancellation context.
```